device/bgp.yaml

```diff
@@ -0,0 +1,87 @@
+components:
+  schemas:
+    Device.Bgp:
+      x-include:
+      - '../common/common.yaml#/components/schemas/Named.Object'
+      x-status: under-review
+      description: >-
+        Container of basic emulated BGP peer settings.
+
+        This container has the following optional sub containers.
+        Device.BgpAdvanced, Device.BgpCapability, Device.BgpSrTePolicy
+        
+        This container has the following optional lists of containers.
+        Device.Bgpv4RouteRange, Device.Bgpv6RouteRange
+      type: object
+      properties:
+        active:
+          description: >-
+            Activates the protocol which means the protocol can be started
+            or stopped.
+          type: boolean
+          default: True
+        router_id:
+          description: >-
+            The IP address of the emulated router
+          type: string
+          format: ipv4
+          default: 0.0.0.0
+        as_type:
+          description: >-
+            The type of BGP autonomous system.
+            External BGP is used for BGP links between two or more
+            autonomous systems (ebgp)
+            Internal BGP is used within a single autonomous system (ibgp).
+          type: string
+          enum:
+          - ibgp
+          - ebgp
+          default: ibgp
+        as_number:
+          description: >-
+            Autonomous System Number (AS number or ASN)
+          type: integer
+          default: 1
+        as_number_width:
+          description: >-
+            The width in bytes of the as_number values.
+            Any as_number values that exceed the width MUST result in an error
+          type: string
+          enum:
+          - two
+          - four
+          default: four
+        as_number_set_mode:
+          description: >-
+            The AS number set mode
+          type: string
+          enum:
+          - do_not_include_as
+          - include_as_seq
+          - include_as_set
+          - include_as_seq_confed
+          - include_as_set_confed
+          - prepend_as_to_first_segment
+          default: do_not_include_as          
+        advanced:
+          $ref: './bgpadvanced.yaml#/components/schemas/Device.BgpAdvanced'
+        capability:
+          $ref: './bgpcapability.yaml#/components/schemas/Device.BgpCapability'
+        sr_te_policies:
+          description: >-
+            Segment routing/traffic engineering policies
+          type: array
+          items:
+            $ref: './bgpsrtepolicy.yaml#/components/schemas/Device.BgpSrTePolicy'
+        bgpv4_routes:
```

Comment:
Q: My understanding is that the model in bgp.yaml is included in both BGPv4 and BGPv6 devices, ie it contains the common pieces. Do we have scenarios where a BGPv6 router can advertise BGPv4 routes ? and vice-versa?

Reply:
The initial customer ask is a dual stack scenario where bgpv4/v6 peers are sending both v4/v6 routes.